pkg/crc/api/client/websocket_client.go

```diff
@@ -0,0 +1,38 @@
+package client
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/crc-org/crc/pkg/crc/machine/types"
+	"nhooyr.io/websocket"
+	"nhooyr.io/websocket/wsjson"
+)
+
+type WebSocketClient struct {
+	client *http.Client
+}
+
+type StatusCallback func(*types.WmStatusResult, error)
+
+func NewWebSocketClient(httpClient *http.Client) *WebSocketClient {
+	return &WebSocketClient{
+		httpClient,
+	}
+}
+
+func (c *WebSocketClient) Status(callback StatusCallback) error {
```

Comment:
`func (c *WebSocketClient) Status(callback StatusCallback) (types.WmStatusResult, error)`  without the `for {}` loop would be more consistent with the other 'client' APIs  and would avoid the need for the callback. The `for{}` loop could stay in the `cmd/` package.